### PR TITLE
Implement buy options new design

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/23349

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

https://user-images.githubusercontent.com/48117473/172789807-fd58526b-e15a-43bb-abb4-3bee09467e41.mov

### CODE REVIEW

File: components/brave_wallet_ui/constants/types.ts [hunk 1]
```diff
@@ -673,7 +673,10 @@ export enum CoinTypesMap {
 
 export type BuyOption = {
   id: BraveWallet.OnRampProvider
-  label: string
+  icon: any
```
Comment (josheleonard) [inline components/brave_wallet_ui/constants/types.ts:676]: Should this be a ImageSource or ReactNode?

File: components/brave_wallet_ui/options/buy-with-options.ts [hunk 2]
```diff
@@ -6,13 +6,22 @@
 import { getLocale } from '$web-common/locale'
 import { BraveWallet, BuyOption } from '../constants/types'
 
+import * as RampIcon from '../assets/svg-icons/ramp-icon.svg'
```
Comment (josheleonard) [inline components/brave_wallet_ui/options/buy-with-options.ts:9]: is `* as` needed?

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/buy-send-swap/buy/index.tsx [hunk 3]
```diff
@@ -13,52 +12,57 @@ import { getLocale } from '../../../../common/locale'
...
   } = props
+
+  const [buyAmount, setBuyAmount] = React.useState('')
+  const [showBuyOptions, setShowBuyOptions] = React.useState<boolean>(false)
   const [buyOptions, setBuyOptions] = React.useState<BuyOption[]>(BuyOptions)
 
-  const onShowAssets = () => {
-    onChangeBuyView('assets', 'from')
-  }
+  // Redux
```
[the remark this answers is not in the document]
Comment (muliswilliam) [reply components/brave_wallet_ui/components/buy-send-swap/buy/index.tsx:38]: 🚀

File: components/brave_wallet_ui/constants/types.ts (hunk 1 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/constants/types.ts]: I have changed this to `string`. Image `src` are essentially strings.

File: components/brave_wallet_ui/options/buy-with-options.ts (hunk 2 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/options/buy-with-options.ts]: Refactored

**Comment by kjozwiak** [comment]
@muliswilliam changing the milestone to `1.41.x` as this landed in `master` which is currently `1.41.x` and not `1.40.x`. Please see https://github.com/brave/brave-browser/wiki/Triage-Guidelines#uplifting 👍

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.41.61 Chromium: 103.0.5060.42 (Official Build) nightly (64-bit) 
-- | --
Revision | `de0d840bf9439c31bd86bf74f065c31fdf9b208d-refs/branch-heads/5060@{#667}`
OS | Windows 11 Version 21H2 (Build 22000.708)

- Verified steps from PR
- Verified a new button to `Select purchase method...` is shown on the Buy widget
- Verified selecting it shows available providers
- Verified both `Ramp` and `Wyre` is shown for available tokens/network
- Verified only `Ramp` is shown for tokens/network when applicable

https://user-images.githubusercontent.com/17010094/173501596-abf43069-2157-4529-a3ec-2095270ba375.mp4

**Pull request merged**